R/summarize_scores.R

```diff
@@ -50,14 +49,12 @@ summarize_scores.data.frame <- function(data, weights = .risk_weights) {
 
 #' @export
 summarize_scores.list <- function(data, weights = .risk_weights) {
```

Comment:
I'd suggest adding the default value to add a default weight to our internal .risk_weights within the summarize_score function. This meets our needs for two different behaviors
1. When no weights are provided, fill in 1s for all missing columns, meaning we don't have to keep `.risk_weights` up-to-date as we add new metrics unless we want a weight other than 1.
2. Allows users to specify a partial set of weights, in which case missing columns will be unused (weight of 0)

```suggestion


add_default_weights <- function(data, weights) {
  cols_missing_weights <- setdiff(names(data), names(weights))
  weights[cols_missing_weights] <- 1
  weights 
}


#' @export
summarize_scores.list <- function(data, weights) {
  # when no weights are provided, add in a weight of 1 for all non-specified columns
  if (is.missing(weights)) weights <- add_default_weights(data, .risk_weights)
```

Reply:
Hi @dgkf, thank you for reviewing the PR. I have a couple of questions.
1. Is `is.missing` a function in an R package? If not, we can set the weights arguments to `NULL` by default and check if it so in the body of the function:
   ```r
    summarize_scores.list <- function(data, weights = NULL) {
     # when no weights are provided, add in a weight of 1 for all non-specified columns
     if (is.null(weights)) weights <- add_default_weights(data, .risk_weights)
   ```

2. Should we also add these modifications to the `summarize_scores.data.frame` function?

3. Can we remove the `.risk_weights` list? If the user doesn't provide any weights, we could set the weight of each metric to 1. If the user does provide weights for some metrics, we could use those and ignore any missing metrics (weight = 0).

Reply:
> 1. Is is.missing a function in an R package

My mistake - the base function is just `missing()`. It returns `TRUE` if no argument is passed to this parameter (even if a default is available) and `FALSE` otherwise. I always mix this one up.

> 2. Should we also add these modifications to the summarize_scores.data.frame function?

Yes, another oversight on my part. Thanks for catching that.

> 3. Can we remove the .risk_weights list?

Yeah, I think we could. The original idea was that we might want to provide a default weighting convention that might be more involved than just giving everything a weight of 1. We haven't really made major progress toward that, though, and I think it might be misleading users to think that we have some profound meaning behind our suggested weights when they're just placeholder. Let's get rid of it and we can reintroduce it if we establish a more meaningful set of suggested weights.

Reply:
great, thanks @dgkf! I'll make those changes soon.